interfaces/Portalicious/src/app/components/query-table/components/query-table-global-search/query-table-global-search.component.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  input,
+  model,
+  output,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+
+import { ButtonModule } from 'primeng/button';
+import { IconFieldModule } from 'primeng/iconfield';
+import { InputIconModule } from 'primeng/inputicon';
+import { InputTextModule } from 'primeng/inputtext';
+
+@Component({
+  selector: 'app-query-table-global-search',
+  standalone: true,
+  imports: [IconFieldModule, InputIconModule, ButtonModule, InputTextModule],
+  templateUrl: './query-table-global-search.component.html',
+  styles: ``,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueryTableGlobalSearchComponent {
+  globalFilterValue = input<string | undefined>();
+  globalFilterVisible = model<boolean>(false);
+  readonly onFilterChange = output<string | undefined>();
+
+  @ViewChild('globalFilterContainer')
+  globalFilterContainer: ElementRef<HTMLDivElement>;
+  @ViewChild('globalFilterInput')
+  globalFilterInput: ElementRef<HTMLInputElement>;
+
+  constructor(private renderer: Renderer2) {
+    this.renderer.listen('window', 'click', (e: Event) => {
```

Comment:
This looks very low-level?

Is there no "dismiss-on-backdrop-click" or something?

And would this still listen after the component is gone, or is the renderer keeping track of that?

Also, I think the same would need to happen when pressing "Esc" on the keyboard?

Reply:
All good questions, but out of scope for the PR. I'm just moving code around. That said, here are the answers.

> Is there no "dismiss-on-backdrop-click" or something?

I found this when googling how to do this stuff in Angular. Couldn't find anything pre-made.

> And would this still listen after the component is gone, or is the renderer keeping track of that?

Good question, I'm not sure.

> Also, I think the same would need to happen when pressing "Esc" on the keyboard?

Nobody asked for it, and this has been reviewed and used by everyone, so it looks like nobody needs/wants it.

Reply:
Well,  I think "the team" (in/excluding PO) is in charge of the DoD.

Acceptance Criteria is the area of 'outside stakeholders' and PO's, etc... But let's leave the semantics to Diderik. :D

One of the reasons/requirements of the UI-kit/library we searched for was its "accessibility-level".
And although I don't think "creating an accessible product" is "an absolute thing", it is something to aim for; and at least not "omit something that helps". So I consider it the same as some specific TypeScript/ESLint-toggle; "does is help? (to create a better product)", then we should add/include it. "does it make it worse? (or harder to work with in the future)", then we shouldn't.

Reply:
I don't disagree with you, I'm more concerned with this only being a priority when you look at a PR, and it being neglected when anyone else does.